nameserver/dns.go

```diff
@@ -70,14 +70,14 @@ func (d *DNSServer) listen(address string) error {
 	if err != nil {
 		return err
 	}
-	udpServer := &dns.Server{PacketConn: udpListener, Handler: d.createMux(d.udpClient, minUDPSize)}
+	udpServer := &dns.Server{PacketConn: udpListener, Handler: d.createMux(d.udpClient, minUDPSize).mux}
```

Comment:
The `.mux` is a wart. I'd lift the mux creation out of `createMux`, and make that function, suitably renamed, just add the handlers.